plugins/woocommerce/includes/admin/class-wc-admin-settings.php

```diff
@@ -796,20 +799,24 @@ public static function save_fields( $options, $data = null ) {
 
 			// Loop options and get values to save.
 			foreach ( $options as $option ) {
-				if ( ! isset( $option['id'] ) || ! isset( $option['type'] ) || ( isset( $option['is_option'] ) && false === $option['is_option'] ) ) {
+				if( isset( $option['name'] ) ) {
+					$option['id'] = $option['name'];
+				}
```

Comment:
Thanks for updating this, but I don't think it solves the problem [described here](https://github.com/woocommerce/woocommerce/pull/32728/files#issuecomment-1130574544) (you can try it out by repeating the same test), because at this point `$option['name']` will not be set for most default options.

Reply:
Thank you for the review!
I didn't understand why it didn't work on your side, but I completely forgot to commit the rest of the file before pushing it... lol
Sorry for that

Reply:
OK, thanks!

This works. However, I am concerned that mutating `$option['id']` may lead to confusion for other developers. 

For example, suppose a new setting has been registered where the `name` and `ID` are two different things: it would be quite reasonable to expect **no change** to `$options['ID']` when inspecting it via [`woocommerce_update_option_<TYPE>`](https://github.com/woocommerce/woocommerce/pull/32728/files#diff-ad6c64c89daea54321e8585e99172fc86844d7c62798418f91d421c5398f826eR870) (or any of the other hooks in this method).

How about using a new var for this—or perhaps restructure to use existing var `$option_name`—and leave the `$option` array itself untouched?